mypy/test/helpers.py

```diff
@@ -367,6 +367,10 @@ def assert_type(typ: type, value: object) -> None:
 def parse_options(program_text: str, testcase: DataDrivenTestCase,
                   incremental_step: int, based: bool = False) -> Options:
     """Parse comments like '# flags: --foo' in a test case."""
+    import mypy.options
+    # This is extremely sus as it's a global option shared by all tests.
+    #  But it seems to be okay (I tested it)
+    mypy.options._based = based
```

Comment:
whh is this sus check needed if its already checked in a less sus way here https://github.com/KotlinIsland/basedmypy/pull/309/files#diff-a1728e8ba6e78bdc975a04007a7765fc56c65959bef3a8ca87b94fbf85370572R128

Reply:
```py
options = parse_options(original_program_text, testcase, incremental_step, based=based)
```

What you linked to is where it figures out if it's based or not, this code is where it turns it on.